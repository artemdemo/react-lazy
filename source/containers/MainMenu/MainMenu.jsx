import React from 'react';
import Link from 'react-router/lib/Link';
import Container from '../../components/Container/Container';

import './MainMenu.less';

const MainMenu = () => {
    return (
        <div className='main-menu'>
            <Container>
                <div className='main-menu-list'>
                    <div className='main-menu-list-item'>
                        <div className='main-menu-list-item__content'>
                            <Link to='/'>Dashboard</Link>
                        </div>
                    </div>
                    <div className='main-menu-list-item'>
                        <div className='main-menu-list-item__content'>
                            <Link to='/orders'>Orders</Link>
                        </div>
                    </div>
                    <div className='main-menu-list-item
                                    main-menu-list-item_parent'>
                        <div className='main-menu-list-item__content
                                        main-menu-list-item__content_has-submenu'>
                            Stores
                        </div>
                        <div className='main-menu-sublist'>
                            <div className='main-menu-sublist__item'>
                                <Link to='/stores/all'>All</Link>
                            </div>
                            <div className='main-menu-sublist__item'>
                                <Link to='/stores/new-york'>New York</Link>
                            </div>
                            <div className='main-menu-sublist__item'>
                                <Link to='/stores/tel-aviv'>Tel Aviv</Link>
                            </div>
                            <div className='main-menu-sublist__item'>
                                <Link to='/stores/tokyo'>Tokyo</Link>
                            </div>
                        </div>
                    </div>
                    <div className='main-menu-list-item'>
                        <div className='main-menu-list-item__content'>
                            <Link to='/cms'>CMS</Link>
                        </div>
                    </div>
                </div>
            </Container>
        </div>
    );
};

export default MainMenu;
