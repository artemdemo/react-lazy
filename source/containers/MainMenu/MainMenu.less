@import '../../styles/functions';

@border-color: #e4e4e4;

.main-menu {
    border-top: 1px solid @border-color;
    border-bottom: 1px solid @border-color;
    margin-bottom: 10px;
    box-shadow: 0 1px 5px rgba(0, 0, 0, 0.06);
}

.main-menu-list {
    .clearFix();
}

.main-menu-list-item {
    float: left;
    border-left: 1px solid @border-color;

    &:first-child {
        border-left: none;
    }
}
    .main-menu-list-item_parent {
        cursor: default;

        &:hover .main-menu-sublist {
            display: block;
        }
    }

    .main-menu-list-item__content {
        padding: 8px 10px;
    }
        .main-menu-list-item__content_has-submenu {
            position: relative;
            padding-right: 25px;

            &:after {
                font-family: 'Glyphicons Halflings';
                content: '\e259';
                font-size: 80%;
                position: absolute;
                right: 5px;
                top: 10px;
            }
        }

.main-menu-sublist {
    display: none;
    position: absolute;
    background-color: white;
    border-left: 1px solid @border-color;
    border-right: 1px solid @border-color;
    box-shadow: 0 6px 10px rgba(0, 0, 0, 0.1);
    z-index: 1;

    // aligning border of submenu with parent item
    margin-left: -1px;
}
    .main-menu-sublist__item {
        padding: 5px 10px;
        border-bottom: 1px solid @border-color;
    }
