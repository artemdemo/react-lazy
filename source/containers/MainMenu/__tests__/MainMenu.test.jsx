import React from 'react';
import renderer from 'react-test-renderer';
import MainMenu from '../MainMenu';

describe('<MainMenu>', () => {
    it('Render', () => {
        const tree = renderer.create(
            <MainMenu />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
