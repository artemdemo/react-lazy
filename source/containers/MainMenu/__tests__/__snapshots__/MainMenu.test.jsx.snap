// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainMenu> Render 1`] = `
<ul>
  <li>
    <a
      onClick={[Function]}
      style={Object {}}
    >
      Main page
    </a>
  </li>
  <li>
    <a
      onClick={[Function]}
      style={Object {}}
    >
      Second page
    </a>
  </li>
</ul>
`;
